app/models/server/models/RoomEvents.ts

```diff
@@ -0,0 +1,147 @@
+import { IEDataGenesis } from '../../../events/definitions/data/IEDataGenesis';
+import { IEDataMessage } from '../../../events/definitions/data/IEDataMessage';
+import { IEDataUpdate } from '../../../events/definitions/data/IEDataUpdate';
+import { EDataDefinition, EventTypeDescriptor, IEvent } from '../../../events/definitions/IEvent';
+import { IRoom } from '../../../events/definitions/IRoom';
+import { getLocalSrc } from '../../../events/server/lib/getLocalSrc';
+import { AddEventResult, ContextQuery, EventsModel, EventStub } from './Events';
+
+const getContextQuery = (param: string | IEvent<any>): ContextQuery => {
+  let rid: string;
+
+  if (typeof param === 'string') {
+    rid = param;
+  } else {
+    rid = param.rid;
+  }
+
+  return { rid };
+};
+
+class RoomEventsModel extends EventsModel {
+  constructor() {
+    super('message');
+
+    // this.tryEnsureIndex({ 'context.roomId': 1 });
+    // this.tryEnsureIndex({ 'd.msg': 'text' }, { sparse: true });
+  }
+
+  public ensureSrc(src: string) {
+    return src || getLocalSrc();
+  }
+
+  public async addRoomEvent<T extends EDataDefinition>(event: IEvent<T>): Promise<AddEventResult> {
+    return super.addEvent(getContextQuery(event), event);
+  }
+
+  public async updateRoomEventData<T extends EDataDefinition>(event: IEvent<T>, dataToUpdate: IEDataUpdate<T>): Promise<void> {
+    return super.updateEventData(getContextQuery(event), event._cid, dataToUpdate);
+  }
+
+  public async createRoomGenesisEvent(src: string, room: IRoom): Promise<IEvent<IEDataGenesis>> {
+    src = this.ensureSrc(src);
+
+    const event: IEDataGenesis = { room };
+
+    return super.createGenesisEvent(src, getContextQuery(room._id), event);
+  }
+
+  public async createMessageEvent<T extends IEDataMessage>(src: string, roomId: string, _cid: string, d: T): Promise<IEvent<T>> {
+    src = this.ensureSrc(src);
+
+    const stub: EventStub<T> = {
+      _cid,
+      t: EventTypeDescriptor.MESSAGE,
+      d,
+    };
+
+    return super.createEvent(src, getContextQuery(roomId), stub);
+  }
+
+  public async createEditMessageEvent<T extends IEDataMessage>(src: string, roomId: string, _cid: string, d: IEDataUpdate<T>): Promise<IEvent<T>> {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [variable name must be in lowerCamelCase, PascalCase or UPPER_CASE](https://app.codacy.com/app/RocketChat/Rocket-Chat/pullRequest?prid=4420492)